lib/core/realtime.rb

```diff
@@ -52,38 +51,134 @@ def run_bot
     end
 
     @client.on :message do |data|
```

Comment:
This method should likely be refactored to be shorter in some way.  Possibly to call methods for each case in the switch.